tests/chainer_tests/datasets_tests/test_sub_dataset.py

```diff
@@ -0,0 +1,119 @@
+import unittest
+
+from chainer import datasets
+from chainer import testing
+
+
+class TestSubDataset(unittest.TestCase):
+
+    def test_sub_dataset(self):
+        original = [1, 2, 3, 4, 5]
+        subset = datasets.SubDataset(original, 1, 4)
+        self.assertEqual(len(subset), 3)
+        self.assertEqual(subset[0], 2)
+        self.assertEqual(subset[1], 3)
+        self.assertEqual(subset[2], 4)
+
```

Comment:
Please add unittests that access dataset outside of the boundary and check errors are correctly raised.